th2_common/schema/factory/common_factory.py

```diff
@@ -83,23 +90,184 @@ def create_from_arguments(args=None):
+                mq = loads(box_data['mq.json'])
+
+                for grpc_service in grpc['services']:
+
+                    endpoints = grpc['services'][grpc_service]['endpoints']
+
+                    for endpoint in endpoints:
+
+                        endpoint_port = endpoints[endpoint]['port']
+
+                        for service in services_dict['items']:
+
+                            if service['metadata']['name'] == box_name:
+
+                                ports = service['spec']['ports']
+
+                                for port in ports:
+
+                                    if port['target_port'] == endpoint_port:
+                                        grpc['services'][grpc_service]['endpoints'][endpoint]['port'] = \
+                                            port['node_port']
+
+                                        grpc['services'][grpc_service]['endpoints'][endpoint]['host'] = \
+                                            kube_contexts[1]['name']
+                                    break
+                            break
+
+        for cassandra_config_map in config_maps_dict['items']:
+
+            if cassandra_config_map['metadata']['name'] == 'cradle':
+
+                cradle = loads(cassandra_config_map['data']['cradle.json'])
+
+                cradle['host'] = kube_contexts[1]['name']
+
+                for item in service_in_namespace_service['items']:
+
+                    if item['metadata']['name'] == 'cassandra-schema':
+
+                        for port in item['spec']['ports']:
+
+                            if port['port'] == int(cradle['port']):
+                                cradle['port'] = port['node_port']
+                                break
+                break
+
+        for rabbit_config_map in config_maps_dict['items']:
+            if rabbit_config_map['metadata']['name'] == 'rabbit-mq-app-config':
+                rabbit = loads(rabbit_config_map['data']['rabbitMQ.json'])
+                rabbit['host'] = kube_contexts[1]['name']
+                rabbit['username'] = kube_contexts[1]['context']['user']
+
+                for item in service_in_namespace_service['items']:
+
+                    if item['metadata']['name'] == 'rabbitmq-schema':
+
+                        for port in item['spec']['ports']:
+
+                            if port['port'] == int(rabbit['port']):
+                                rabbit['port'] = port['node_port']
+                                break
+                break
+
+        dictionary = CommonFactory.__get_dictionary(box_name, v1.list_config_map_for_all_namespaces())
+
+        try:
+            mkdir(config_dir)
+            LOGGER.info("Directory %s is created at %s" % (config_dir, getcwd()))
+        except OSError:
+            LOGGER.info("All configuration in the %s folder are overridden" % (getcwd() + "/" + config_dir))
+
+        try:
+            grpc_file = open(grpc_path, "w")
+            grpc_file.write(dumps(grpc))
```

Comment:
Here and below, to work with closed resources as files, you must use the *with* instruction.